pkg/services/store/object/httpobjectstore/service.go

```diff
@@ -0,0 +1,369 @@
+package httpobjectstore
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"strconv"
+	"strings"
+
+	"github.com/grafana/grafana/pkg/infra/log"
+	"github.com/grafana/grafana/pkg/middleware"
+	"github.com/grafana/grafana/pkg/services/store/kind"
+	"github.com/grafana/grafana/pkg/services/store/object"
+	"github.com/grafana/grafana/pkg/services/store/router"
+	"github.com/grafana/grafana/pkg/util"
+	"github.com/grafana/grafana/pkg/web"
+
+	"github.com/grafana/grafana/pkg/api/response"
+	"github.com/grafana/grafana/pkg/api/routing"
+	"github.com/grafana/grafana/pkg/models"
+)
+
+type HTTPObjectStore interface {
+	// Register HTTP Access to the store
+	RegisterHTTPRoutes(routing.RouteRegister)
+}
+
+type httpObjectStore struct {
+	store  object.ObjectStoreServer
+	log    log.Logger
+	kinds  kind.KindRegistry
+	router router.ObjectStoreRouter
+}
+
+func ProvideHTTPObjectStore(store object.ObjectStoreServer, kinds kind.KindRegistry) HTTPObjectStore {
+	return &httpObjectStore{
+		store:  store,
+		log:    log.New("http-object-store"),
+		kinds:  kinds,
+		router: router.NewObjectStoreRouter(kinds),
+	}
+}
+
+// All registered under "api/object"
+func (s *httpObjectStore) RegisterHTTPRoutes(route routing.RouteRegister) {
+	// For now, require admin for everything
+	reqGrafanaAdmin := middleware.ReqSignedIn //.ReqGrafanaAdmin
+
+	// Every * must parse to a GRN (uid+kind)
+	route.Get("/store/*", reqGrafanaAdmin, routing.Wrap(s.doGetObject))
```

Comment:
would routing be easier to maintain if we registered the patterns from `RouteFromKey` as path params instead of parsing manually, and then constructed the GRN from the parsed path params (maybe in middleware)?

Reply:
perhaps -- but having a single place to explore this is WAY easier now then trying to mirror the same logic in two places